baseplate/crypto.py

```diff
@@ -92,89 +122,97 @@ class SignatureInfo(_SignatureInfo):
     pass
 
 
-class MessageSigner(object):
-    """Helper which signs messages and validates signatures given a secret.
+def _compute_digest(secret_value, header, message):
+    payload = header + message.encode("utf8")
+    digest = hmac.new(secret_value, payload, hashlib.sha256).digest()  # pylint: disable=no-member
+    return digest
 
-    .. testsetup::
 
-        import datetime
-        from baseplate.crypto import MessageSigner
+def make_signature(secret, message, max_age):
+    """Return a signature for the given message.
 
-    .. testcode::
+    To ensure that key rotation works automatically, always fetch the secret
+    token from the secret store immediately before use and do not cache / save
+    the token anywhere. The ``current`` version of the secret will be used to
+    sign the token.
 
-        message = "Hello, world!"
+    :param baseplate.secrets.VersionedSecret secret: The secret signing key
+        from the secret store.
+    :param str message: The message to sign.
+    :param datetime.timedelta max_age: The amount of time in the future
+        the signature will be valid for.
+    :return: An encoded signature.
 
-        signer = MessageSigner(b"supersecret")
-        signature = signer.make_signature(
-            message, max_age=datetime.timedelta(days=1))
+    """
+    version = 1
+    expiration = int(time.time() + max_age.total_seconds())
```

Comment:
Mayyyybe, but if so I'd want you to have to opt into it really noisily. Trying to make it hard to do the wrong thing in the helpers that go into this lib. Do you have any use cases in mind? (And perhaps let's keep that for another PR since this is just reworking the helpers to use versioned secrets?)